community/kernel/src/main/java/org/neo4j/kernel/impl/locking/community/CommunityLockClient.java

```diff
@@ -44,129 +61,243 @@ public CommunityLockClient( LockManagerImpl manager )
     @Override
     public void acquireShared( Locks.ResourceType resourceType, long resourceId )
     {
-        PrimitiveLongObjectMap<LockResource> localLocks = localShared( resourceType );
-        LockResource resource = localLocks.get( resourceId );
-        if( resource != null )
+        if ( !stateHolder.incrementActiveClients() )
         {
-            resource.acquireReference();
+            return;
```

Comment:
Do the methods not need to communicate if they are unable to acquire their respective locks? Right now they just return immediately as if the lock was acquired.